pkg/backingstore/validation.go

```diff
@@ -54,8 +54,10 @@ func ValidateBSEmptySecretName(bs nbv1.BackingStore) error {
 	switch bs.Spec.Type {
 	case nbv1.StoreTypeAWSS3:
 		if len(bs.Spec.AWSS3.Secret.Name) == 0 {
-			return util.ValidationError{
-				Msg: "Failed creating the Backingstore, please provide secret name",
+			if len(*bs.Spec.AWSS3.AWSSTSRoleARN) == 0 {
```

Comment:
I think it will be best to separate these validation to different functions, to be more accurate with error messages and also to keep function responsibility clear and single as a good design.

Reply:
Seperated into different function